File: src/Model/XmlSequenceWrapper.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using AutoRest.Core.Utilities;

namespace AutoRest.Java.Model
{
    public class XmlSequenceWrapper
    {
        private readonly ListType sequenceType;
        private readonly string xmlRootElementName;
        private readonly string xmlListElementName;
        private readonly string wrapperClassName;
        private readonly ISet<string> imports;

        public XmlSequenceWrapper(ListType sequenceType, string xmlRootElementName, string xmlListElementName, ISet<string> imports)
        {
            this.sequenceType = sequenceType;
            this.xmlRootElementName = xmlRootElementName;
            this.xmlListElementName = xmlListElementName;
            this.wrapperClassName = xmlRootElementName.ToPascalCase() + "Wrapper";
            this.imports = imports;
        }

        public ListType SequenceType => sequenceType;

        public string XmlRootElementName => xmlRootElementName;

        public string XmlListElementName => xmlListElementName;

        public string WrapperClassName => wrapperClassName;

        public ISet<string> Imports => imports;
    }
}

File: javagen/src/main/java/com/azure/autorest/model/clientmodel/XmlSequenceWrapper.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.azure.autorest.model.clientmodel;

import java.util.Set;

/**
 * The details needed to create an XML sequence wrapper class for the client.
 */
public class XmlSequenceWrapper {
    private final String packageName;
    private final IType sequenceType;
    private final String xmlRootElementName;
    private final String xmlListElementName;
    private final String wrapperClassName;
    private final Set<String> imports;

    public XmlSequenceWrapper(String packageName, IType sequenceType, String modelTypeName, String xmlRootElementName,
        String xmlListElementName, Set<String> imports) {
        this.packageName = packageName;
        this.sequenceType = sequenceType;
        this.xmlRootElementName = xmlRootElementName;
        this.xmlListElementName = xmlListElementName;
        this.wrapperClassName = modelTypeName + "Wrapper";
        this.imports = imports;
    }

    public final String getPackage() {
        return packageName;
    }

    public final IType getSequenceType() {
        return sequenceType;
    }

    public final String getXmlRootElementName() {
        return xmlRootElementName;
    }

    public final String getXmlListElementName() {
        return xmlListElementName;
    }

    public final String getWrapperClassName() {
        return wrapperClassName;
    }

    public final Set<String> getImports() {
        return imports;
    }
}
